File: solutions/typescript/darts/1/darts.ts
export function score(x: number, y: number): number {
  const r = (x ** 2 + y ** 2) ** 0.5;
  if (r > 10) return 0;
  if (r <= 10 && r > 5) return 1;
  if (r <= 5 && r > 1) return 5;
  return 10;
}

File: solutions/haskell/darts/1/src/Darts.hs
module Darts (score) where

score :: Float -> Float -> Int
score x y
  | r <= 1 = 10
  | r <= 5 = 5
  | r <= 10 = 1
  | otherwise = 0
  where r = distance x y



-- `^` is for Integer and `**` is for Float
distance :: Float -> Float -> Float
distance x y = sqrt (x**2 + y**2)

